### ISSUE TITLE
b-vae data not inverse transformed.

### ISSUE THREAD

**Issue by errorterror6** [errorterror6/ML_charge_modeling#22]
Yes, there is a significant normalization inconsistency that could cause issues.

  In the SHJNN library, the training data is normalized using StandardScaler:
  # In libs/shjnn/data.py
  data_scaler = preprocessing.StandardScaler().fit(np.concatenate(processed_data))
  normalized_data = [data_scaler.transform(device_data) for device_data in processed_data]

  However, during evaluation in parameter_tuning/b_vae.py, the model predictions are compared against raw,
  non-normalized data without inverse transforming the outputs:
  - No inverse transformation is applied to model outputs
  - MSE is calculated directly between normalized predictions and non-normalized ground truth

  This inconsistency leads to:
  1. Incorrect loss calculation
  2. Misleading visualizations
  3. Interpretation problems with the model outputs

  To fix this, you should apply inverse_transform using the same scaler before evaluation.

> then why does inference for the purpose of plotting appear to work fine? because the evaluation uses infer_step.
  does infer_step output the inverse transformed data, or does the model output inverse transformed data by
  defualt?

● Good question. Let me investigate how the inference and plotting actually work to understand this apparent
  contradiction.

● Task(Examine the inference and plotting code to understand why visualization seems to work despite the …
      normalization inconsistency. Specifically:…)
  ⎿  Done (10 tool uses · 38.4k tokens · 46.5s)

● The visualization appears to work despite the normalization inconsistency for two main reasons:

  1. Consistent scaling during visualization: In parameter_tuning/visualisation.py, both the original data and
  model predictions are scaled by the same factor when plotting:
  axes[dim].plot(orig_time, orig_traj[:, dim] / scale_factor, '.', alpha=0.6, color=color)
  axes[dim].plot(pred_times - 1.0, pred_x_np[:, dim] / scale_factor, '-', linewidth=2, alpha=0.4, color=color)
  2. Manual scaling during data loading: In parameter_tuning/loader.py, data is manually scaled:
  n['data'][j,1][::_ds]*1e2  # response, scale to ~one

  The model implicitly learns to output values in the normalized space it was trained on. When visualized, both
  predictions and ground truth are divided by the same scale factor, making them visually comparable despite being
  in different numerical ranges.

  However, this doesn't fix the evaluation issue - MSE calculations would still be incorrect when comparing
  normalized predictions against non-normalized ground truth.
